packages/client/hooks/useTransition.ts

```diff
@@ -80,16 +80,19 @@ const useTransition = <T extends {key: Key}>(children: T[]) => {
     const mountingKeys = [] as Key[]
     children.forEach((nextChild) => {
       const idxInPrev = prevTChildren.findIndex(({child}) => child.key === nextChild.key)
-      const status = idxInPrev === -1 ? TransitionStatus.MOUNTED : prevTChildren[idxInPrev]!.status
+      const needToUpdateStatus =
+        idxInPrev === -1 || prevTChildren[idxInPrev]!.status === TransitionStatus.EXITING
```

Comment:
Could you provide repro steps so that I can see what bug this fixes? As `useTransition` is being used in many files, it would be great if we can solve this issue without updating `useTransition`